### PR TITLE
[LLM] - [LIVE-10674] - Fix favorite currencies not loading when out of marketData first page

### PR DESCRIPTION
**Pull request by sshmaxime**
### 📝 Description

This PR includes a fix that was preventing user's favorite currencies to not load when out of the `marketData` first page.

https://github.com/LedgerHQ/ledger-live/assets/25589782/3e3df451-d158-4b1c-b531-e4021b089722

### ❓ Context

- **JIRA or GitHub link**: [LIVE-10674] 

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-10674]: https://ledgerhq.atlassian.net/browse/LIVE-10674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/Market/index.tsx [hunk 1]
```diff
@@ -289,6 +289,12 @@ export default function Market({ navigation }: NavigationProps) {
     if (!isConnected) setIsLoading(false);
   }, [isConnected]);
 
+  useEffect(() => {
+    if (starredMarketCoins.length > 0) {
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/Market/index.tsx:293]: Is it possible to to this only if "Starred" Filter is on?

**Comment by KVNLS** [comment]
@sshmaxime can you apply the fix on the newArch too, please?

**Comment by mcayuelas-ledger** [comment]
You can also check why this useEffect not working correctly 
https://github.com/LedgerHQ/ledger-live/pull/5798/files

**Comment by sshmaxime** [comment]
Done @KVNLS

**Comment by alalmi-ledger** [comment]
QA OK for me, cannot repro the issue with a build from the branch ✅

**Review by mcayuelas-ledger — APPROVED** [review]

**Pull request merged**